executor/show_test.go

```diff
@@ -104,9 +104,9 @@ func (s *testSuite5) TestShowWarnings(c *C) {
 	tk.MustExec("set @@sql_mode=''")
 	tk.MustExec("insert show_warnings values ('a')")
 	c.Assert(tk.Se.GetSessionVars().StmtCtx.WarningCount(), Equals, uint16(1))
-	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|1292|Truncated incorrect FLOAT value: 'a'"))
```

Comment:
I found out that this is not compatible with MySQL either...

```sql
drop table if exists t;
set @@sql_mode='';
create table t (a int);
insert into t values ('a');
show warnings;
```

```
+---------+------+------------------------------------------------------+
| Level   | Code | Message                                              |
+---------+------+------------------------------------------------------+
| Warning | 1366 | Incorrect integer value: 'a' for column 'a' at row 1 |
+---------+------+------------------------------------------------------+
```

Reply:
@tangenta Yes, you are right,  insert sentence do have different warning/error information, i have already create a new issue, see #24927